app/models/reporting/temporal_report.rb

```diff
@@ -77,13 +77,19 @@ def previous_date(_date)
       end
       # :nocov:
 
-      def from_param(report_type:, period:)
-        date = Date.strptime(period, self::PARAM_FORMAT)
-        new(report_type:, date:)
+      def from_param(report_type:, period:, interval:)
+        klass = klass_for_interval(interval)
```

Comment:
Spoke about on call. This logic was previously in the controller, moved to super class